consensus/state.go

```diff
@@ -1328,6 +1336,13 @@ func (cs *ConsensusState) finalizeCommit(height int64) {
 
 	fail.Fail() // XXX
 
+	// update mempool filter because consensus params might have changed
+	cs.mempool.SetFilter(func(tx types.Tx) bool {
```

Comment:
Is the idea behind using a filter here that we can more easily use any consensus param, rather than just the maxbytes? The only other thing I can imagine us doing prelaunch is enforcing a minfee. I think both of these would be more well suited inside of the mempool itself. This would also avoid us having to use the mutex on the consensus param everytime a new tx is added. I think this wall cause blocking issues, plus the defer is a performance hit.

Reply:
> more well suited inside of the mempool itself

but consensus params can be changed in any block (https://tendermint.com/docs/app-dev/abci-spec.html#endblock). you mean we should just copy them over if they change. That actually not a bad idea.